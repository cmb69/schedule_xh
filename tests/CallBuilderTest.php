<?php

namespace Schedule;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeRequest;
use Plib\View;

class CallBuilderTest extends TestCase
{
    public function testRendersCallBuilder(): void
    {
        $sut = new CallBuilder(
            "./plugins/schedule/",
            XH_includeVar("./config/config.php", "plugin_cf")["schedule"],
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["schedule"])
        );
        Approvals::verifyHtml($sut(new FakeRequest())->output());
    }
}
