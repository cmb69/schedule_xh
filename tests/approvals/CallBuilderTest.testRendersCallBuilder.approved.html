
<script type="module" src="/plugins/schedule/schedule.js?&amp;v=2.1-dev"></script>
<h1>Schedule – Call Builder</h1>
<form class="schedule_call_builder">
  <p>
    <label>
      <span>Name</span>
      <input name="name" required pattern="[a-z0-9\-]+">
    </label>
  </p>
  <p>
    <label>
      <span>Show totals</span>
      <input name="show_totals" type="checkbox" >
    </label>
  </p>
  <p>
    <label>
      <span>Read only</span>
      <input name="read_only" type="checkbox" >
    </label>
  </p>
  <p>
    <label>
      <span>Multi</span>
      <input name="multi" type="checkbox" checked>
    </label>
  </p>
  <p>
    <label>
      <span>Options</span>
      <textarea name="options" required></textarea>
    </label>
  </p>
  <p class="schedule_buttons">
    <button type="button" name="parse">↑ Parse</button>
    <button type="button" name="build">↓ Build</button>
  </p>
  <p>
    <label>
      <span>Plugin call</span>
      <textarea name="plugin_call"></textarea>
    </label>
  </p>
</form>
